Xext/security.c

```diff
@@ -430,6 +435,15 @@ ProcSecurityGenerateAuthorization(ClientPtr client)
         return BadValue;
     }
 
+    if (client->swapped) {
+        swaps(&stuff->nbytesAuthProto);
+        swaps(&stuff->nbytesAuthData);
+        swapl(&stuff->valueMask);
```

Comment:
These values are used above, we probably should do the swapping before anything is used